### PR TITLE
Added distro selection feature

### PR DESCRIPTION
**Pull request by imabedalghafer**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [x ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
VM repair

**Comment by imabedalghafer** [comment]
@zhoxing-ms , Would you please let us know the update for the review of this PR ?

File: src/vm-repair/azext_vm_repair/_help.py [hunk 1]
```diff
@@ -22,6 +22,9 @@
         - name: Create a repair VM and set the VM authentication
           text: >
             az vm repair create -g MyResourceGroup -n myVM --repair-username username --repair-password password!234 --verbose
+        - name: Create a repair VM of a specific distro , a specific URN could also be provided
```
Comment (zhoxing-ms) [inline src/vm-repair/azext_vm_repair/_help.py:25]:
```suggestion
        - name: Create a repair VM of a specific distro or a specific URN could also be provided
```

File: src/vm-repair/azext_vm_repair/repair_utils.py [hunk 2]
```diff
@@ -407,6 +429,116 @@ def _fetch_compatible_windows_os_urn(source_vm):
...
+    elif distro == 'sles15':
+        os_image_urn = _suse_image_selector('sles-15')
+    else:
+        if distro.count(":") == 3:
+            logger.info('A custom URN was provided , will be used as distro for the recovery VM')
+            os_image_urn = distro
+        else:
+            logger.info('No specific distro was provided , using the default Ubuntu distro')
+            os_image_urn = "UbuntuLTS"
+    return os_image_urn
```
Comment (zhoxing-ms) [inline src/vm-repair/azext_vm_repair/repair_utils.py:496]: Could we configure these mapping relationships into a `dict` to replace the logic of `if-else`

File: src/vm-repair/azext_vm_repair/repair_utils.py [hunk 3]
```diff
@@ -407,6 +429,116 @@ def _fetch_compatible_windows_os_urn(source_vm):
...
+            logger.info('A custom URN was provided , will be used as distro for the recovery VM')
+            if distro.find('gen2'):
+                os_image_urn = distro
+            else:
+                logger.info('The provided URN does not contain Gen2 in it and this VM is a gen2 , dropping to default image')
+                os_image_urn = "Canonical:UbuntuServer:18_04-lts-gen2:latest"
+        else:
+            logger.info('No specific distro was provided , using the default Ubuntu distro')
+            os_image_urn = "Canonical:UbuntuServer:18_04-lts-gen2:latest"
+    return os_image_urn
```
Comment (zhoxing-ms) [inline src/vm-repair/azext_vm_repair/repair_utils.py:539]: Same as above

Comment (imabedalghafer) [reply src/vm-repair/azext_vm_repair/repair_utils.py:539]: Thank you for the update and notes :) 
sure, I will go over the comments and do the  needed (Y)

File: src/vm-repair/azext_vm_repair/repair_utils.py (hunk 2 above)
Comment (imabedalghafer) [reply src/vm-repair/azext_vm_repair/repair_utils.py]: the ```dict```  has been updated to replace the main ```if-else``` logic

File: src/vm-repair/azext_vm_repair/_help.py (hunk 1 above)
Comment (imabedalghafer) [reply src/vm-repair/azext_vm_repair/_help.py]: updated

**Comment by zhoxing-ms** [comment]
@imabedalghafer Could you please resolve those CI issues?

**Comment by wangzelin007** [comment]
> @imabedalghafer Could you please resolve those CI issues?

```
/home/vsts/work/1/s/src/vm-repair/azext_vm_repair/repair_utils.py:471:5: E122 continuation line missing indentation or outdented
/home/vsts/work/1/s/src/vm-repair/azext_vm_repair/repair_utils.py:471:14: E231 missing whitespace after ':'
/home/vsts/work/1/s/src/vm-repair/azext_vm_repair/repair_utils.py:472:5: E122 continuation line missing indentation or outdented
/home/vsts/work/1/s/src/vm-repair/azext_vm_repair/repair_utils.py:472:14: E231 missing whitespace after ':'
/home/vsts/work/1/s/src/vm-repair/azext_vm_repair/repair_utils.py:473:5: E122 continuation line missing indentation or outdented
/home/vsts/work/1/s/src/vm-repair/azext_vm_repair/repair_utils.py:473:14: E231 missing whitespace after ':'
/home/vsts/work/1/s/src/vm-repair/azext_vm_repair/repair_utils.py:474:5: E122 continuation line missing indentation or outdented
/home/vsts/work/1/s/src/vm-repair/azext_vm_repair/repair_utils.py:474:14: E231 missing whitespace after ':'
/home/vsts/work/1/s/src/vm-repair/azext_vm_repair/repair_utils.py:475:5: E122 continuation line missing indentation or outdented
/home/vsts/work/1/s/src/vm-repair/azext_vm_repair/repair_utils.py:475:13: E231 missing whitespace after ':'
/home/vsts/work/1/s/src/vm-repair/azext_vm_repair/repair_utils.py:476:5: E122 continuation line missing indentation or outdented
/home/vsts/work/1/s/src/vm-repair/azext_vm_repair/repair_utils.py:476:13: E231 missing whitespace after ':'
/home/vsts/work/1/s/src/vm-repair/azext_vm_repair/repair_utils.py:494:16: E231 missing whitespace after ':'
/home/vsts/work/1/s/src/vm-repair/azext_vm_repair/repair_utils.py:495:16: E231 missing whitespace after ':'
/home/vsts/work/1/s/src/vm-repair/azext_vm_repair/repair_utils.py:496:16: E231 missing whitespace after ':'
/home/vsts/work/1/s/src/vm-repair/azext_vm_repair/repair_utils.py:497:19: E231 missing whitespace after ':'
/home/vsts/work/1/s/src/vm-repair/azext_vm_repair/repair_utils.py:498:19: E231 missing whitespace after ':'
/home/vsts/work/1/s/src/vm-repair/azext_vm_repair/repair_utils.py:499:18: E231 missing whitespace after ':'
/home/vsts/work/1/s/src/vm-repair/azext_vm_repair/repair_utils.py:500:18: E231 missing whitespace after ':'
/home/vsts/work/1/s/src/vm-repair/azext_vm_repair/repair_utils.py:501:18: E231 missing whitespace after ':'
/home/vsts/work/1/s/src/vm-repair/azext_vm_repair/repair_utils.py:502:18: E231 missing whitespace after ':'
/home/vsts/work/1/s/src/vm-repair/azext_vm_repair/repair_utils.py:503:18: E231 missing whitespace after ':'
/home/vsts/work/1/s/src/vm-repair/azext_vm_repair/repair_utils.py:504:18: E231 missing whitespace after ':'
/home/vsts/work/1/s/src/vm-repair/azext_vm_repair/repair_utils.py:505:17: E231 missing whitespace after ':'
/home/vsts/work/1/s/src/vm-repair/azext_vm_repair/repair_utils.py:506:17: E231 missing whitespace after ':'
/home/vsts/work/1/s/src/vm-repair/azext_vm_repair/repair_utils.py:507:9: E123 closing bracket does not match indentation of opening bracket's line
/home/vsts/work/1/s/src/vm-repair/azext_vm_repair/repair_utils.py:602:9: F402 import 'log' from line 6 shadowed by loop variable
13    E122 continuation line missing indentation or outdented
1     E123 closing bracket does not match indentation of opening bracket's line
1     E128 continuation line under-indented for visual indent
26    E231 missing whitespace after ':'
1     F402 import 'log' from line 6 shadowed by loop variable
```

**Comment by imabedalghafer** [comment]
> > @imabedalghafer Could you please resolve those CI issues?
> 
> ```
> /home/vsts/work/1/s/src/vm-repair/azext_vm_repair/repair_utils.py:471:5: E122 continuation line missing indentation or outdented
> /home/vsts/work/1/s/src/vm-repair/azext_vm_repair/repair_utils.py:471:14: E231 missing whitespace after ':'
> /home/vsts/work/1/s/src/vm-repair/azext_vm_repair/repair_utils.py:472:5: E122 continuation line missing indentation or outdented
> /home/vsts/work/1/s/src/vm-repair/azext_vm_repair/repair_utils.py:472:14: E231 missing whitespace after ':'
> /home/vsts/work/1/s/src/vm-repair/azext_vm_repair/repair_utils.py:473:5: E122 continuation line missing indentation or outdented
> /home/vsts/work/1/s/src/vm-repair/azext_vm_repair/repair_utils.py:473:14: E231 missing whitespace after ':'
> /home/vsts/work/1/s/src/vm-repair/azext_vm_repair/repair_utils.py:474:5: E122 continuation line missing indentation or outdented
> /home/vsts/work/1/s/src/vm-repair/azext_vm_repair/repair_utils.py:474:14: E231 missing whitespace after ':'
> /home/vsts/work/1/s/src/vm-repair/azext_vm_repair/repair_utils.py:475:5: E122 continuation line missing indentation or outdented
> /home/vsts/work/1/s/src/vm-repair/azext_vm_repair/repair_utils.py:475:13: E231 missing whitespace after ':'
> /home/vsts/work/1/s/src/vm-repair/azext_vm_repair/repair_utils.py:476:5: E122 continuation line missing indentation or outdented
> /home/vsts/work/1/s/src/vm-repair/azext_vm_repair/repair_utils.py:476:13: E231 missing whitespace after ':'
> /home/vsts/work/1/s/src/vm-repair/azext_vm_repair/repair_utils.py:494:16: E231 missing whitespace after ':'
> /home/vsts/work/1/s/src/vm-repair/azext_vm_repair/repair_utils.py:495:16: E231 missing whitespace after ':'
> /home/vsts/work/1/s/src/vm-repair/azext_vm_repair/repair_utils.py:496:16: E231 missing whitespace after ':'
> /home/vsts/work/1/s/src/vm-repair/azext_vm_repair/repair_utils.py:497:19: E231 missing whitespace after ':'
> /home/vsts/work/1/s/src/vm-repair/azext_vm_repair/repair_utils.py:498:19: E231 missing whitespace after ':'
> /home/vsts/work/1/s/src/vm-repair/azext_vm_repair/repair_utils.py:499:18: E231 missing whitespace after ':'
> /home/vsts/work/1/s/src/vm-repair/azext_vm_repair/repair_utils.py:500:18: E231 missing whitespace after ':'
> /home/vsts/work/1/s/src/vm-repair/azext_vm_repair/repair_utils.py:501:18: E231 missing whitespace after ':'
> /home/vsts/work/1/s/src/vm-repair/azext_vm_repair/repair_utils.py:502:18: E231 missing whitespace after ':'
> /home/vsts/work/1/s/src/vm-repair/azext_vm_repair/repair_utils.py:503:18: E231 missing whitespace after ':'
> /home/vsts/work/1/s/src/vm-repair/azext_vm_repair/repair_utils.py:504:18: E231 missing whitespace after ':'
> /home/vsts/work/1/s/src/vm-repair/azext_vm_repair/repair_utils.py:505:17: E231 missing whitespace after ':'
> /home/vsts/work/1/s/src/vm-repair/azext_vm_repair/repair_utils.py:506:17: E231 missing whitespace after ':'
> /home/vsts/work/1/s/src/vm-repair/azext_vm_repair/repair_utils.py:507:9: E123 closing bracket does not match indentation of opening bracket's line
> /home/vsts/work/1/s/src/vm-repair/azext_vm_repair/repair_utils.py:602:9: F402 import 'log' from line 6 shadowed by loop variable
> 13    E122 continuation line missing indentation or outdented
> 1     E123 closing bracket does not match indentation of opening bracket's line
> 1     E128 continuation line under-indented for visual indent
> 26    E231 missing whitespace after ':'
> 1     F402 import 'log' from line 6 shadowed by loop variable
> ```

Apologies for the delay , sure I will work on them

**Comment by imabedalghafer** [comment]
@wangzelin007, I have resolved the above CLI errors. would you please check

**Comment by imabedalghafer** [comment]
@wangzelin007 Any update please ?

**Comment by wangzelin007** [comment]
> @wangzelin007 Any update please ?

Sorry for the late reply, I will merge this PR ASAP.

**Comment by wangzelin007** [comment]
> @wangzelin007 Any update please ?

Hi [imabedalghafer](https://github.com/imabedalghafer),

May I ask whether it is necessary to add a scenario test for this new argument ?

**Comment by imabedalghafer** [comment]
@wangzelin007 Thank you for the update.
let me check internally on that part and will get back to you

**Comment by imabedalghafer** [comment]
@wangzelin007 
Apologies for the delay
The test units, history and setup files are updated 

Thank you and much appreciate your help

**Review by wangzelin007 — APPROVED** [review]

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**